rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -411,20 +411,39 @@ impl Controller {
         Ok(())
     }
 
-    fn reload_resource_list(&mut self) -> Result<()> {
-        let Some(session) = &self.session else {
-            tracing::warn!("got notified to update resources but there is no session");
-            return Ok(());
-        };
-        let resources = session.callback_handler.resources.load();
-        let actor_name = &self
-            .auth
-            .session()
-            .ok_or_else(|| anyhow!("connlib is signed in but auth module isn't"))?
-            .actor_name;
-        self.app
-            .tray_handle()
-            .set_menu(system_tray_menu::signed_in(actor_name, &resources))?;
+    /// Returns a new system tray menu
+    fn build_system_tray_menu(&self) -> Result<tauri::SystemTrayMenu> {
+        Ok(if let Some(auth_session) = self.auth.session() {
+            if let Some(connlib_session) = &self.session {
+                // TODO: Modify CallbackHandler so it can tell us whether the tunnel is really ready.
+                let tunnel_ready = true;
+                if tunnel_ready {
+                    // Signed in, tunnel ready
+                    let resources = connlib_session.callback_handler.resources.load();
+                    system_tray_menu::signed_in(&auth_session.actor_name, &resources)
+                } else {
+                    // Signed in, raising tunnel
+                    system_tray_menu::signing_in()
+                }
+            } else {
+                bail!("We have an auth session but no connlib session");
```

Comment:
I think this should be a `debug_assert`

Reply:
I made it a `tracing::error`, I want it to be in release logs since it's a logic error that's cheap to find, but yeah maybe it shouldn't panic in release.